sanity/lib/utils.ts

```diff
@@ -1,10 +1,35 @@
-import client from './sanity';
-import getImage from './getImage';
+import { createClient } from 'next-sanity';
+
+import { config } from './config';
+
+export const client = createClient(config);
+
+//* HOME PAGE *\\
+export const fetchSpecificSkills = async <T extends String>(
+  skill: T | T[]
+) => {
+  const fetchedSkills = await client.fetch<any[]>(
+    `*[_type == "skills" && name ${
+      Array.isArray(skill) && !(skill instanceof String)
+        ? `in [${skill.map((skill) => `"${skill}"`).join(', ')}]`
+        : `match ${skill.trim()}`
+    }]{
+      ...,
+      "ref": icon.asset._ref
+    }`
+  );
+
+  if (!fetchedSkills) {
+    console.error('Error fetching those skill', fetchedSkills);
+    return null;
+  }
+
+  return fetchedSkills;
+};
 
 //* PROJECT PAGE *\\
-export const fetchProjects = await client.fetch<any[]>(
-  `*[_type == "projects"]`
-);
+export const fetchProjects = async () =>
+  await client.fetch<any[]>(`*[_type == "projects"]`);
```

Comment:
```suggestion
  await client.fetch(`*[_type == "projects"]`);
```
Generically typing it as `any[]` is the same as just not including a generic type. If you want to take advantage of typing with generics- then you need to define a type that matches what you're requesting back (this is just an example, change it depending on the specific shape of data you're expecting back):

```js
interface ProjectCardInfo {
  name: string;
  description: string;
  img: string;
}
//* PROJECT PAGE *\\
export const fetchProjects = async () =>
  await client.fetch<ProjectCardInfo[]>(`*[_type == "projects"]{
    name,
    description,
    "img": imgUrl.asset->url
  }`);
  ```
  
  then you would get the desired code editor completion tools: 
  
![image](https://github.com/jacastanon01/portfolio/assets/4108484/a8d2a942-8f6b-4c3c-b80e-be0d925d95e3)

again this is just an example.